spring-sync-core/src/main/java/org/springframework/sync/FromOperation.java

```diff
@@ -39,4 +41,25 @@ public String getFrom() {
 		return from;
 	}
 
+	@Override
+	public boolean equals(Object o) {
+		if (o instanceof FromOperation) {
+			return super.equals(o) && Objects.equals(from, ((FromOperation) o).from);
+		}
+		return false;
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(op, path, value, from);
+	}
+
+	@Override
+	public String toString() {
+		return "PatchOperation{" +
```

Comment:
```suggestion
		return "FromOperation{" +
```
?

Reply:
`getClass().getSimpleName()` can be used